src/services/middlewareServices/AuthorizationMiddlewareService.ts

```diff
@@ -0,0 +1,58 @@
+import { NotFoundError } from "@errors/NotFoundError"
+
+import UserWithSiteSessionData from "@classes/UserWithSiteSessionData"
+
+import { E2E_ISOMER_ID } from "@root/constants"
+import AuthService from "@services/identity/AuthService"
+import IsomerAdminsService from "@services/identity/IsomerAdminsService"
+import UsersService from "@services/identity/UsersService"
+
+// Import logger
+const logger = require("@logger/logger")
+
+interface AuthorizationMiddlewareServiceProps {
+  identityAuthService: AuthService
+  usersService: UsersService
+  isomerAdminsService: IsomerAdminsService
+}
+
+export default class AuthorizationMiddlewareService {
+  readonly identityAuthService: AuthorizationMiddlewareServiceProps["identityAuthService"]
+
+  readonly usersService: AuthorizationMiddlewareServiceProps["usersService"]
+
+  readonly isomerAdminsService: AuthorizationMiddlewareServiceProps["isomerAdminsService"]
+
+  constructor({
+    identityAuthService,
+    usersService,
+    isomerAdminsService,
+  }: AuthorizationMiddlewareServiceProps) {
+    this.identityAuthService = identityAuthService
+    this.usersService = usersService
+    this.isomerAdminsService = isomerAdminsService
+  }
+
+  async checkIsSiteMember(sessionData: UserWithSiteSessionData) {
+    // Check if user has access to site
+    const { siteName, isomerUserId: userId } = sessionData
+
+    // Should always be defined - authorization middleware only exists if siteName is defined
+    if (!siteName) throw Error("No site name in authorization middleware")
+
+    logger.info(`Verifying user's access to ${siteName}`)
+
+    const isSiteMember = await (sessionData.isEmailUser()
+      ? this.usersService.hasAccessToSite(userId, siteName)
+      : this.identityAuthService.hasAccessToSite(sessionData))
+
+    const isAdminUser = await this.isomerAdminsService.getByUserId(userId)
+
+    const isE2EUser = userId === E2E_ISOMER_ID
+    if (!isSiteMember && !isAdminUser && !isE2EUser) {
+      throw new NotFoundError("Site does not exist")
```

Comment:
Hmm I think we're setting it this way so we don't give out additional information, especially since we're planning to swap over to displaying error messages directly from the backend - a user should not be able to distinguish between a site that he does not have access to and one that doesn't exist (which i think is the same strategy that github applies)

Reply:
tl;dr from offline discussion: we want _authorization_ errors to show up as missing sites. this is how github does it at present and we want to emulate that behaviour so that we don't leak info (that the user is not authorized to view the site)